File: week6/py/maximum_product.py
#/* https://www.geeksforgeeks.org/breaking-integer-to-get-maximum-product/ */

n=int(input())
if (n == 2 or n == 3):
	print(n - 1); 
else:
    res = 1; 
    while (n > 4):
        n -= 3
        res *= 3
    print(n * res)

File: week6/cpp/maximum_product.cpp
/* https://www.geeksforgeeks.org/breaking-integer-to-get-maximum-product/ */
#include <iostream>
using namespace std;

int main() {
    int n;
    cin >> n;

    if (n == 2 || n == 3) {
        cout << n - 1 << endl;
    } else {
        int res = 1;

        while (n > 4) {
            n -= 3;
            res *= 3;
        }

        cout << n * res << endl;
    }

    return 0;
}
